admin/cli/cli.go

```diff
@@ -151,12 +151,19 @@ func printResponse(opts *flags.GlobalFlags, res commands.Result, err error) erro
 
 func printSuccessResult(opts *flags.GlobalFlags, res commands.Result) {
 	if opts.JSON {
-		b, jErr := json.Marshal(res)
-		if jErr != nil {
+		switch res.(type) {
+		case json.Marshaler:
+			b, jErr := json.Marshal(res)
```

Comment:
if `res` is a `json.Marshaler`, insn't there a `res.Marshal` instead of calling `json.Marshal(res)`?